src/UnitOfWork.php

```diff
@@ -3019,36 +3014,26 @@ public function getScheduledCollectionUpdates(): array
      */
     public function initializeObject(object $obj): void
     {
-        if ($obj instanceof InternalProxy) {
-            $obj->__load();
-
-            return;
-        }
-
         if ($obj instanceof PersistentCollection) {
             $obj->initialize();
 
             return;
         }
 
-        if ($this->em->getConfiguration()->isNativeLazyObjectsEnabled()) {
-            $reflection = $this->em->getClassMetadata($obj::class)->getReflectionClass();
-            $reflection->initializeLazyObject($obj);
-        }
+        $reflection = $this->em->getClassMetadata($obj::class)->getReflectionClass();
+        $reflection->initializeLazyObject($obj);
     }
 
     /**
      * Tests if a value is an uninitialized entity.
-     *
-     * @phpstan-assert-if-true InternalProxy $obj
```

Comment:
this assertion should already be removed in 3.4.x, as it is not true when enabling native lazy objects.

Reply:
Done in https://github.com/doctrine/orm/pull/12033